src/map/lua/luautils.cpp

```diff
@@ -903,7 +903,16 @@ namespace luautils
         // clang-format off
         lua.set_function("GetFirstID", [&](std::string const& name) -> uint32
         {
-            return lookup[name].front();
+            if (lookup.find(name) != lookup.end())
+            {
+                return lookup[name].front();
+            }
+            else
+            {
+                ShowWarning(fmt::format("GetFirstID({}): Returning nil", name));
```

Comment:
This error message could have some more context. Also, it ends up returning 0, not nil.

Reply:
Useful to remember that the only things in Lua that are falsey are `false` and `nil` (`if 0 then print("zero") end` will print, etc.)

Reply:
Yea probably a good idea to provide the path of the `IDs.lua` that is being processed for both this and `GetTableOfIDs`

Reply:
Updated warning messages with the zone name in all permutations of failures for both functions:

![image](https://github.com/LandSandBoat/server/assets/131182600/7983430c-a5da-4f25-9aa4-92958782100b)

I can't actually figure out how to trigger this warning, but updated as well:

![image](https://github.com/LandSandBoat/server/assets/131182600/458ca467-0d51-4582-9d45-5c4396c0b914)